modules/develop/pages/data-transforms/how-transforms-work.adoc

```diff
@@ -0,0 +1,41 @@
+= How Data Transforms Work
+:description: Learn how Redpanda data transforms work.
+
+Redpanda provides the framework to build and deploy inline transformations on data written to Redpanda topics, delivering processed and validated data to consumers in the format they expect. Redpanda does this directly inside the broker, eliminating the need to manage a separate stream processing environment or use 3rd-party tools. 
+
+image::shared:wasm1.png[Data transforms in a broker] 
+
+To learn how to build and deploy data transforms, see xref:./data-transforms.adoc[].
+
+== Wasm in Redpanda
+
+Data transforms use a WebAssembly (Wasm) engine inside a Redpanda broker, allowing Redpanda to control the entire transform lifecycle. For example, Redpanda can stop and start transforms when partitions are moved or to free up system resources for other tasks. 
+
+Within Repdanda, there's a single Raft controller that manages cluster information, including data transforms. On every shard (core), Redpanda knows what data transforms exist in the cluster, as well as metadata about the transform, such as input and output topics and environment variables. 
+
+Wasm is executed within a runtime in the broker. Data transforms take an input topic and map it to an output topic, running a virtual machine (VM) on the same shard as the input partition leader. Redpanda uses just-in-time (JIT) compilation, which compiles the function in memory, writes it to executable space, then runs the directly translated machine code. When it becomes leader of a given partition on the input topic, Redpanda does the following:
```

Comment:
>  running a virtual machine (VM) on the same shard as the input partition leader

We should make it clear that a transform runs on the same shard as _every partition leader_ for a topic.

Reply:
> When it becomes leader of a given partition on the input topic, Redpanda does the following:

When a shard becomes the leader...